From b333464ba4d18b10e70543100655373c9037fd15 Mon Sep 17 00:00:00 2001
From: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
Date: Thu, 9 Nov 2023 14:17:05 +0530
Subject: [PATCH] ath12k:Fixing the crash in MBSSID bring-up

This crash is seen in latest rebased code changes when configuring
MBSSID interface. This is due to usage of virtual interface structure
without link ID. Since all EHT hwmode need proper link ID for interface
bring-up and updated the structure in the ath12k module.

Signed-off-by: Gautham Kumar Senthilkumaran <quic_gauthamk@quicinc.com>
Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
Signed-off-by: Aaradhana Sahu <quic_aarasahu@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.h | 2 +-
 drivers/net/wireless/ath/ath12k/wmi.c  | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.h b/drivers/net/wireless/ath/ath12k/core.h
index 9127330..9c5a843 100644
--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -458,7 +458,7 @@ struct ath12k_vif {
 struct ath12k_vif_iter {
 	u32 vdev_id;
 	struct ath12k *ar;
-	struct ath12k_vif *arvif;
+	struct ath12k_link_vif *arvif;
 };
 
 #define HAL_AST_IDX_INVALID	0xFFFF
diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index 1c387b7..dbd576a 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -8439,7 +8439,7 @@ ath12k_wmi_fw_vdev_stats_fill(struct ath12k *ar,
 	if (!arvif)
 		return;
 
-	vif_macaddr = arvif->vif->addr;
+	vif_macaddr = arvif->addr;
 
 	len += scnprintf(buf + len, buf_len - len, "%30s %u\n",
 			 "VDEV ID", vdev->vdev_id);
@@ -8525,7 +8525,7 @@ ath12k_wmi_fw_bcn_stats_fill(struct ath12k *ar,
 {
 	u32 len = *length;
 	u32 buf_len = ATH12K_FW_STATS_BUF_SIZE;
-	struct ath12k_vif *arvif = ath12k_mac_get_arvif(ar, bcn->vdev_id);
+	struct ath12k_link_vif *arvif = ath12k_mac_get_arvif(ar, bcn->vdev_id);
 	u8 *vdev_macaddr;
 
 	if (!arvif) {
@@ -8534,7 +8534,7 @@ ath12k_wmi_fw_bcn_stats_fill(struct ath12k *ar,
 		return;
 	}
 
-	vdev_macaddr = arvif->vif->addr;
+	vdev_macaddr = arvif->addr;
 
 	len += scnprintf(buf + len, buf_len - len, "%30s %u\n",
 			 "VDEV ID", bcn->vdev_id);
-- 
2.17.1

